Add script to run all MPM test cases

akturner opened the pull request:
Added script to run all the MPM test cases at once 
Run model scripts now assume standard location of executable if MPAS_SEAICE_EXECUTABLE is undefined 
Add log message script to test case utils
When all test cases run a summary log file is generated 
Log file has ANSI text codes for colored text when viewed in more, cat, tail etc

sulsky reviewed (commented):
I think it would be useful to have
        print("Using executable: %s" %(MPAS_SEAICE_EXECUTABLE))
print at the beginning of the summary log file, log_mpm_testcases.txt.

On perlmutter, the run command is going to involve srun, not mpirun, so we are going to need some flexibility here. (We could wait on this until we are ready for perlmutter testing.)

Maybe we could check for the presence of a log.seaice.0000.err file in each current testcase directory as a way of telling whether or not the testcase ran?  

I pointed to an executable that did not have the config parameter config_use_mpm_freeze_melt in the registry.  So, all testcases failed to run with an error CRITICAL ERROR: Error while reading namelist record mpm in the log.seaice.0000.err file.
But the log_mpm_testcases.txt file only said:

MPAS-Seaice-MPM test cases
##########################

Advection test case
===================
   Elapsed time: 1.2740 seconds

Advection convergence test case
===============================
   Elapsed time: 16.2682 seconds

Assembly test case
==================
   Elapsed time: 0.7484 seconds

Column on particles test case
=============================
   Elapsed time: 0.8069 seconds

MPM tracers test case
=====================
   Elapsed time: 0.7670 seconds

Polynya test case
=================
Check plots
  /home/sulsky/fortran/MPAS-SI-MPM/test/scriptmods/MPAS-Seaice_standalone_fr
amework/testcases/MPM/polynya/cells_scatter.png
  /home/sulsky/fortran/MPAS-SI-MPM/test/scriptmods/MPAS-Seaice_standalone_fr
amework/testcases/MPM/polynya/particles_scatter.png
   Elapsed time: 195.8964 seconds

Interpolation test case
=======================
   Elapsed time: 101.4750 seconds

Reconstruction test case
========================
   Elapsed time: 101.8632 seconds

Standard physics test case
==========================
   Elapsed time: 2.0930 seconds
Testing Completed

akturner commented:
@sulsky: I added better checking for if the model actually ran, and the script reports what executable it is going to use. I moved the actual execution of the model into a separate function so this can be modified in a follow up PR to make the testing work fully on perlmutter

sulsky reviewed (commented):
Maybe just stop execution and ask for MPAS_SEAICE_TESTCASES_RUN_COMMAND to be set instead of:

Traceback (most recent call last):
  File "/home/sulsky/fortran/MPAS-SI-MPM/test/scriptmods/MPAS-Seaice_standalone_framework/testcases/MPM/column_on_particles/run_testcase.py", line 260, in <module>
    run_testcase(args.nProcs,
  File "/home/sulsky/fortran/MPAS-SI-MPM/test/scriptmods/MPAS-Seaice_standalone_framework/testcases/MPM/column_on_particles/run_testcase.py", line 152, in run_testcase
    execute_model(MPAS_SEAICE_EXECUTABLE,
  File "/home/sulsky/fortran/MPAS-SI-MPM/test/scriptmods/MPAS-Seaice_standalone_framework/testcases/MPM/column_on_particles/../../../utils/testcases/execute_model.py", line 19, in execute_model
    raise Exception("Unsupported MPAS_SEAICE_TESTCASES_RUN_COMMAND type: %s" %(MPAS_SEAICE_TESTCASES_RUN_COMMAND))
Exception: Unsupported MPAS_SEAICE_TESTCASES_RUN_COMMAND type: None

Would you set up spherical_operators/strain and spherical_operators/strain_stress_divergence to be run if desired?

Thanks!

sulsky commented:
My other comment did not seem to post ...
Can remove:  https://github.com/akturner/MPAS-Seaice_standalone_framework/blob/c33ee895d533ce11fd92486f7dd5cf92a53918b2/testcases/MPM/run_testcases.py#L25
